src/structure/wavelettree.rs

```diff
@@ -196,30 +196,24 @@ impl WaveletTree {
     }
 }
 
-fn build_wavelet_fragment<
-    S: Stream<Item = u64, Error = std::io::Error> + Send,
-    W: AsyncWrite + Send,
->(
+fn build_wavelet_fragment<S: Stream<Item = u64, Error = std::io::Error>, W: AsyncWrite>(
     stream: S,
     write: BitArrayFileBuilder<W>,
     alphabet: usize,
     layer: usize,
     fragment: usize,
-) -> impl Future<Item = BitArrayFileBuilder<W>, Error = std::io::Error> + Send {
+) -> impl Future<Item = BitArrayFileBuilder<W>, Error = std::io::Error> {
```

Comment:
This is the one thing I am unsure of. By removing the Send trait, this future is no longer spawnable from tokio. The fact that you were able to do this seems to indicate that we weren't actually ever doing that, which seems very odd to me.

Reply:
Since it's easy enough to reintroduce the trait bound I'll still merge though.